File: HelloWorld/app/src/main/java/henryxi/com/MainActivity.java
package henryxi.com;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import javax.security.auth.login.LoginException;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.i(TAG, "onCreate: after create");
    }
}

File: chapter04/UICustomViews/app/src/main/java/com/example/uicustomviews/MainActivity.kt
package com.example.uicustomviews

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //todo 文本的图片可以显示,但是button的背景图显示不出来,怀疑和theme有关.
        supportActionBar?.hide()
    }
}